File: gas_station.py
class Solution:
    def canCompleteCircuit(self, gas, cost):
        sum = _sum = start = 0
        for i in range(0, len(gas), 1):
            sum += gas[i] - cost[i]
            _sum += gas[i] - cost[i]
            if sum < 0:
                sum = 0
                start = i + 1

        if _sum < 0:
            return -1
        return start

File: dp/gas_station.cpp
#include <iostream>

using namespace std;

int main(void) {

    int gas[10] = {1,3,3,5,1,7,2,3,4,5};
    int cost[10] = {1,1,3,3,2,4,2,3,1,2};
    int tank = 1; // assume start from station 0
    for (int i = 1;i < 10; i++) {
        if (tank >= cost[i]) {
            tank = tank - cost[i];
            tank += gas[i];
        }
        else
            break;
        cout<<"tank:"<<tank<<" next station cost:"<<cost[i]<<endl;

    }


    return 0;
}
